### PR TITLE
Added rubies/ruby/CVE-2017-17790.yml

### PR DESCRIPTION
**Pull request by jasnow**
See if I have fixed my branching issues.

Adding a single file, rubies/ruby/CVE-2017-17790.yml file.

It is related to issue #438.

Feel free to edit the file. Some of the text came from the GHSA database.

### CODE REVIEW

**Comment by jasnow** [comment]
Verified all green checks.

File: rubies/ruby/CVE-2017-17790.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+engine: ruby
+cve: 2017-17790
+url: https://nvd.nist.gov/vuln/detail/CVE-2017-17790
+title: The lazy_initialize function in lib/resolv.rb in Ruby...
+date: 2017-12-20
+description: |
+  "The lazy_initialize function in lib/resolv.rb in Ruby through 2.4.3
```
Comment (postmodern) [inline rubies/ruby/CVE-2017-17790.yml:8]: Can remove the lone double quote.

File: rubies/ruby/CVE-2017-17790.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+engine: ruby
+cve: 2017-17790
+url: https://nvd.nist.gov/vuln/detail/CVE-2017-17790
+title: The lazy_initialize function in lib/resolv.rb in Ruby...
```
Comment (postmodern) [inline rubies/ruby/CVE-2017-17790.yml:5]: Can probably remove the `...` from the title.

File: rubies/ruby/CVE-2017-17790.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  character, a different vulnerability than CVE-2017-17405.
+  NOTE: situations with untrusted input may be highly unlikely.
+
+  CWE: CWE-74 - Improper Neutralization of Special Elements
+
+  CVSS_V3: 9.8 - CRITICAL - CVSS:3.0/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H
+
+cvss_v3: 9.8
+patched_versions:
+ - '>= 2.2.8'
```
Comment (postmodern) [inline rubies/ruby/CVE-2017-17790.yml:20]: Should be `~> 2.2.8` so the version ranges don't overlap.

File: rubies/ruby/CVE-2017-17790.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ - '>= 2.3.5'
```
Comment (postmodern) [inline rubies/ruby/CVE-2017-17790.yml:21]: Should be `~> 2.3.5` so the version ranges don't overlap.

**Review by postmodern — CHANGES_REQUESTED** [review]
Minor changes requested.

**Comment by jasnow** [comment]
Think I fixed all feedback.

**Comment by jasnow** [comment]
Think #438 is done.

**Pull request merged**